infrastructure-formula/infrastructure/libvirt/domains.sls

```diff
@@ -67,6 +68,55 @@ vm_uuid_map_{{ machine }}:
   - name: /etc/uuidmap
   - text: '{{ machine }}: {{ salt['cmd.run']('uuidgen') }}'
   - unless: 'grep -q {{ machine }} /etc/uuidmap'
+
+{%- if clusterpillar[cluster].get('storage') == 'local' and 'image' in config %}
+{%- if 'root' in config['disks'] %}
+{%- set root_disk = topdir ~ '/disks/' ~ machine ~ '_root.qcow2' %}
+{%- set image = topdir ~ '/os-images/' ~ config['image'] %}
+write_vmdisk_{{ machine }}_root:
+  file.copy:
```

Comment:
`file.copy´ is a salt module, maybe you want to use the `file.copy_` state instead? (don't blame me for that naming ;-)

If so, you can replace the `- unless` with `- force`

Reply:
The example says usage is `file.copy`, albeit the state module being `file.copy_`:

https://docs.saltproject.io/en/latest/ref/states/all/salt.states.file.html#salt.states.file.copy_

Reply:
Ah, but it should not overwrite. Thanks, will try.

Reply:
Replaced the condition, works.